features/cellular/framework/AT/AT_CellularDevice.h

```diff
@@ -20,28 +20,26 @@
 
 #include "CellularDevice.h"
 
-#include "AT_CellularNetwork.h"
-#include "AT_CellularSIM.h"
-#include "AT_CellularSMS.h"
-#include "AT_CellularPower.h"
-#include "AT_CellularInformation.h"
-
-#include "ATHandler.h"
-
 namespace mbed {
 
+class ATHandler;
+class AT_CellularInformation;
+class AT_CellularNetwork;
+class AT_CellularPower;
+class AT_CellularSIM;
+class AT_CellularSMS;
+
 /**
  *  Class AT_CellularDevice
  *
  *  A class defines opening and closing of cellular interfaces.
  *  Deleting/Closing of opened interfaces can be done only through this class.
  */
 class AT_CellularDevice : public CellularDevice {
-public:
+protected:
```

Comment:
Do you really need to make the constructor protected here? Is the class really totally unusable as-is until you override something? The fact that you've provided all the default `open_xxx_impl` functions makes it look as if it would be usable as-is.

If it isn't useable as-is, maybe you don't need the default implementation of everything.

Reply:
That was left as a mistake. Now it's back to public.